dashboard/app/models/levels/applab.rb

```diff
@@ -125,12 +129,25 @@ def update_json_fields
   end
 
   def validate_maker_if_needed
-    maker_enabled = properties['makerlab_enabled'] == 'true'
+    maker_api_property = properties['makerlab_enabled']
+    maker_enabled = SharedConstants::MAKER_PALETTE_CATEGORIES.keys.include?(maker_api_property)
     starting_category = properties['palette_category_at_start']
-    if Applab.maker_palette_categories.include?(starting_category) && !maker_enabled
+
+    # If the "Maker" palette category is selected, either Maker API must be enabled.
+    if starting_category == 'Maker' && !maker_enabled
```

Comment:
I think this is easier for me to reason about if the first check handles the cases where a level does not have maker enabled, and has any of the three hardware-related starting categories (not just Maker). And the end of the error message could be "...but this level does not have a Maker API enabled".

Then, the second check handles the specifics of picking a circuit or microbit starting category (which you're already doing).

Reply:
Alright, I went ahead and switched around that first check. Let me know if that reads better.

Reply:
Looks great, thanks!